File: app/src/main/java/com/commit451/gitlab/api/request/SessionRequest.java
package com.commit451.gitlab.api.request;

import com.bluelinelabs.logansquare.annotation.JsonField;
import com.bluelinelabs.logansquare.annotation.JsonObject;

/**
 * Start a session
 */
@JsonObject
public class SessionRequest {

    @JsonField(name = "email")
    String email;
    @JsonField(name = "login")
    String login;
    @JsonField(name = "password")
    String password;

    public SessionRequest setEmail(String email) {
        this.email = email;
        return this;
    }

    public SessionRequest setLogin(String login) {
        this.login = login;
        return this;
    }

    public SessionRequest setPassword(String password) {
        this.password = password;
        return this;
    }
}

File: app/src/main/java/com/commit451/gitlab/api/request/SessionRequest.kt
package com.commit451.gitlab.api.request

import com.squareup.moshi.Json

/**
 * Start a session
 */
class SessionRequest {

    @field:Json(name = "email")
    var email: String? = null
    @field:Json(name = "login")
    var login: String? = null
    @field:Json(name = "password")
    var password: String? = null
}
